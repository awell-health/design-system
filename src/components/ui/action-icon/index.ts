export { ActionIcon } from './action-icon';
