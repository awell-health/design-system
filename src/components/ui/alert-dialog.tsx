import * as React from "react";
interface Props {
  title: string;
  text?: string;
  children?: JSX.Element | string;
  button?: {
    label: string;
    buttonAction?: () => void;
  };
}

const AlertDialog = (props: Props): React.JSX.Element => {
  const { title, text, button, children = null } = props;

  return (
    <div
      className="relative z-10"
      aria-labelledby="modal-title"
      role="dialog"
      aria-modal="true"
    >
      <div className="fixed inset-0 bg-black bg-opacity-80 transition-opacity"></div>
      <div className="fixed inset-0 z-10 w-screen overflow-y-auto">
        <div className="flex min-h-full items-end justify-center p-4 sm:items-center sm:p-0">
          <div className="relative transform overflow-hidden rounded-lg bg-white shadow-xl transition-all sm:my-8 sm:w-full sm:max-w-lg">
            <div className="bg-white p-6 pb-8">
              <div>
                <div>
                  <h3
                    className="text-base font-semibold leading-6 text-gray-900"
                    id="modal-title"
                  >
                    {title}
                  </h3>
                  {text && (
                    <div className="mt-2">
                      <p className="text-sm text-gray-500">{text}</p>
                    </div>
                  )}
                </div>
              </div>
            </div>
            {children && <div className="pb-8 px-6">{children}</div>}
            {button && (
              <div className="px-6 pb-6 flex justify-end">
                <button
                  type="button"
                  className="inline-flex justify-center rounded-md bg-blue-600 px-3 py-2 text-sm font-semibold text-white shadow-sm hover:bg-blue-500 sm:ml-3"
                  onClick={button?.buttonAction}
                >
                  {button?.label}
                </button>
              </div>
            )}
          </div>
        </div>
      </div>
    </div>
  );
};

export { AlertDialog }
