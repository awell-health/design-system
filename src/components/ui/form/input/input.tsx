import * as React from 'react';

import { cn } from '@/lib/utils';

interface InputProps extends React.InputHTMLAttributes<HTMLInputElement> {
  label?: string;
  helpText?: string;
  suffixIcon?: JSX.Element;
  prefixIcon?: JSX.Element;
  hasError?: boolean;
  onChange?: React.ChangeEventHandler<HTMLInputElement>;
  className?: string;
  type?: string;
  labelClassName?: string;
}
const Input = React.forwardRef<HTMLInputElement, InputProps>(
  (
    {
      className,
      label,
      helpText,
      suffixIcon,
      prefixIcon,
      hasError,
      onChange,
      type = 'text',
      labelClassName,
      ...props
    },
    ref
  ) => {
    const renderIcon = (icon: JSX.Element, position: 'left' | 'right') => {
      return (
        <span
          className={cn(
            'absolute h-[40px] flex items-center',
            position === 'left' && 'left-2.5',
            position === 'right' && 'right-2.5'
          )}
        >
          {React.cloneElement(icon, {
            size: '16',
            className: hasError ? 'fill-red-600' : 'fill-slate-500'
          })}
        </span>
      );
    };

    return (
      <div className=''>
        {label && (
          <div className={cn('label', labelClassName)}>
            <span className=' label-text text-slate-600 text-sm font-medium'>{label}</span>
          </div>
        )}
        <label className='form-control w-full flex relative'>
          {prefixIcon && renderIcon(prefixIcon, 'left')}
          <input
            type={type}
            className={cn(
              type !== 'file' && 'px-3.5 py-2.5',
              'h-[38px] min-h-[38px] rounded-lg gap-2 w-full border-box',
              'text-sm font-normal',
              'shadow border border-slate-300 text-black placeholder:text-slate-400',
              'focus:outline-0 focus:border focus:border-blue-500',
              'disabled:bg-slate-50 disabled:text-slate-300',
              prefixIcon && 'pl-8',
              suffixIcon && 'pr-8',
              hasError && 'border-red-500 focus:border-red-500',
              type === 'file' && 'file-input file-input-bordered file-input-primary',
              className
            )}
            onChange={onChange}
            ref={ref}
            {...props}
          />
          {suffixIcon && renderIcon(suffixIcon, 'right')}
        </label>
        {helpText && (
          <div className='label'>
            <span
              className={cn(
                'label-text-alt text-slate-500 text-xs font-normal',
                hasError && 'text-red-600'
              )}
            >
              {helpText}
            </span>
          </div>
        )}
      </div>
    );
  }
);
Input.displayName = 'Input';

export { Input };
