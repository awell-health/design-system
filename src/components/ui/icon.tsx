import * as React from "react";
import * as RemixIcon from "@remixicon/react";

export enum IconSize {
  xs = 16,
  s = 20,
  m = 24,
  l = 28,
  xl = 32,
}

export type IconType = keyof typeof RemixIcon;

interface Props {
  icon: IconType;
  size?: IconSize;
  color?: string;
  className?: string;
}

const Icon = (props: Props): React.JSX.Element => {
  const { icon, color = "default", className = "inline", size = IconSize.m } = props

  const IconComponent = RemixIcon[icon]

  return <IconComponent size={size} color={color} className={className} />;
};

export { Icon };
