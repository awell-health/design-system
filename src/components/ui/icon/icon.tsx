import * as React from 'react';
import { ICONS, IconSize, IconType } from './types';

interface Props {
  icon: IconType;
  size?: IconSize;
  color?: string;
  className?: string;
}

const Icon = (props: Props): React.JSX.Element => {
  const { icon, color = 'default', className = 'inline', size = IconSize.m } = props;

  const IconComponent = ICONS[icon];

  return <IconComponent size={size} color={color} className={className} />;
};

export { Icon };
