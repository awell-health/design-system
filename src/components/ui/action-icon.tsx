import * as React from "react"
import { cn } from "@/lib/utils"
import { Icon, IconType } from "./icon"


export enum ActionType {
  ApiCall = "API_CALL",
  ApiCallGraphql = "API_CALL_GRAPHQL",
  Calculation = "CALCULATION",
  Checklist = "CHECKLIST",
  ClinicalNote = "CLINICAL_NOTE",
  Form = "FORM",
  Message = "MESSAGE",
  Plugin = "PLUGIN",
  PushToEmr = "PUSH_TO_EMR",
}

export interface ActionIconProps {
    actionType?: ActionType
    icon_url?: string
    size?: "sm" | "lg"
}

const ActionIcon = React.forwardRef<HTMLDivElement, ActionIconProps>(
  ({ ...props }, _ref) => {
    const TYPES_ICONS_MAP: Record<
      ActionType,
      IconType
    > = {
      [ActionType.ApiCall]: "RiTerminalBoxFill",
      [ActionType.ApiCallGraphql]: "RiTerminalWindowFill",
      [ActionType.Calculation]: "RiCalculatorFill",
      [ActionType.Checklist]: "RiCheckboxMultipleFill",
      [ActionType.ClinicalNote]:  "RiHealthBookFill",
      [ActionType.Form]: "RiSurveyFill",
      [ActionType.Message]: "RiMessage3Fill",
      [ActionType.Plugin]: "RiUploadCloudFill",
      [ActionType.PushToEmr]: "RiUploadCloudFill"
    }

    const { icon_url, actionType, size='sm' } = props

    const icon =
      TYPES_ICONS_MAP[actionType as keyof typeof TYPES_ICONS_MAP] ??
      "RiQuestionFill"

    const color = icon === "RiQuestionFill" ? "fill-slate-400" : "fill-slate-800";
    const iconSize = size === 'sm' ? 'w-4 h-4' : 'w-6 h-6'
    const containersize = size === 'sm' ? 'w-6 h-6' : 'w-9 h-9'

    if (actionType === ActionType.Plugin && icon_url) {
      return (
        <img
          className={`${containersize} rounded-md border border-slate-200`}
          src={icon_url}
        />
      )
    }

    return (
      <div className={`${containersize} flex bg-slate-100 rounded-md border border-slate-200 items-center justify-center`}>
        <Icon icon={icon} className={cn(`${iconSize}`, color)} />
      </div>
    )
  },
)
ActionIcon.displayName = "ActionIcon"

export { ActionIcon }
