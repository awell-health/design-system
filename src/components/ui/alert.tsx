import * as React from "react";
import { cva, type VariantProps } from "class-variance-authority";

import { cn } from "@/lib/utils";
import { Icon } from "./icon";

const alertVariants = cva("alert p-4 rounded-lg border gap-3", {
  variants: {
    variant: {
      default: "bg-slate-50 border-slate-300 text-slate-600 [&h3]:text-slate-700",
      primary: "bg-blue-50 border-blue-300 text-blue-600 [&h3]:text-blue-700",
      warning: "bg-yellow-50 border-yellow-300 text-yellow-600 [&h3]:text-yellow-700",
      error: "bg-red-50 border-red-300 text-red-600 [&h3]:text-red-700",
    },
  },
  defaultVariants: {
    variant: "default",
  },
});

interface Props
  extends React.HTMLAttributes<HTMLDivElement>,
    VariantProps<typeof alertVariants> {
  title: string;
  showIcon?: boolean
}

const Alert = React.forwardRef<HTMLDivElement, Props>(
  ({ className, variant, title, children, showIcon = false, ...props }, ref) => {
    const icon = variant === 'error' || variant === 'warning' ? 'RiAlertLine' : 'RiInformationLine'
    const iconClassNames = cn({
      'fill-slate-500': variant === 'default',
      'fill-blue-500': variant === 'primary',
      'fill-yellow-500': variant === 'warning',
      'fill-red-500': variant === 'error'
    })

    return (
      <div
        role="alert"
        ref={ref}
        className={cn(alertVariants({ variant }), className)}
        {...props}
      >
        <div className="flex self-start pt-[5px]">
          {showIcon && <Icon icon={icon} size={16} className={iconClassNames} />}
        </div>
        <div>
          <h3 className="font-medium">{title}</h3>
          {children && <div className="font-normal leading-tight text-xs">{children}</div>}
        </div>
      </div>
    )
  }
);

export { Alert };
