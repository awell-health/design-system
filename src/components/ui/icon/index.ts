import { Icon } from './icon';
import { IconType, IconSize } from './types';

export { Icon, type IconType, IconSize };
