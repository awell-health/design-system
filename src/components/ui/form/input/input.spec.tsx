import { render } from '@testing-library/react';
import { expect, it, describe, vi } from 'vitest';
import { Input } from './input';
import { Icon } from '@/components/ui/icon';

describe('Input', () => {
  const subject = (props = {}) => {
    const defaultProps = {
      label: 'Label',
      helpText: 'Help text',
      prefixIcon: <Icon icon='RiAddFill' />,
      suffixIcon: <Icon icon='RiArrowLeftLine' />,
      onChange: vi.fn()
    };

    return render(<Input {...defaultProps} {...props} />);
  };
  it('match snapshot', () => {
    expect(subject().container).toMatchSnapshot();
  });

  it('displays error', () => {
    const value = 'test input value';
    const { getByDisplayValue } = subject({ hasError: true, value });

    expect(getByDisplayValue(value)).toHaveClass('border-red-500');
  });
});
