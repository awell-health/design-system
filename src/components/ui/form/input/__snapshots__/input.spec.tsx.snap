// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Input > match snapshot 1`] = `
<div>
  <div
    class=""
  >
    <div
      class="label flex flex-col gap-1 items-start"
    >
      <span
        class=" label-text text-slate-600 text-sm font-medium"
      >
        Label
      </span>
    </div>
    <label
      class="form-control w-full flex relative"
    >
      <span
        class="absolute h-[40px] flex items-center left-2.5"
      >
        <svg
          class="remixicon fill-slate-500"
          fill="default"
          height="16"
          viewBox="0 0 24 24"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M11 11V5H13V11H19V13H13V19H11V13H5V11H11Z"
          />
        </svg>
      </span>
      <input
        class="px-3.5 py-2.5 h-[38px] min-h-[38px] rounded-lg gap-2 w-full text-sm font-normal shadow border border-slate-300 text-black placeholder:text-slate-400 focus:outline-0 focus:border focus:border-blue-500 disabled:bg-slate-50 disabled:text-slate-300 pl-8 pr-8"
        type="text"
      />
      <span
        class="absolute h-[40px] flex items-center right-2.5"
      >
        <svg
          class="remixicon fill-slate-500"
          fill="default"
          height="16"
          viewBox="0 0 24 24"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M7.82843 10.9999H20V12.9999H7.82843L13.1924 18.3638L11.7782 19.778L4 11.9999L11.7782 4.22168L13.1924 5.63589L7.82843 10.9999Z"
          />
        </svg>
      </span>
    </label>
    <div
      class="label"
    >
      <span
        class="label-text-alt text-slate-500 text-xs font-normal"
      >
        Help text
      </span>
    </div>
  </div>
</div>
`;
